app/helpers/event_helper.rb

```diff
@@ -33,7 +33,9 @@ def property_select_options(url_params, aggregation, props)
 
   def date_select_options(url_params, aggregation, chart_type)
     Event::DATE_OPTIONS.map do |k, v|
-      selected = "selected" if k == url_params[:date]
+      if (!url_params[:date] && k == "7d") || k == url_params[:date]
+        selected = "selected"
+      end
       "<option data-url='#{build_date_url(url_params.permit(*Event::EVENT_PARAMS), aggregation, k, chart_type)}' data-name='#{k}' #{selected}>#{v}</option>"
```

Comment:
Probably good idea to initialize `selected` outside of the scope of the if condition (i.e. before it). I think in this case it always will be defined, but in general such a situation can lead to an error where `selected` is undefined.

Reply:
The issue with the first condition (`!url_params[:date] && k == "7d"`) is that if somebody puts `?date=blabla`, it will never be true and the second condition will also not be true, and selected will be undefined.

Therefore, it would be better to check if url_params[:date] contains a value which is in the keys of `Event::DATE_OPTIONS`.